cpp/src/slice2cpp/Gen.cpp

```diff
@@ -436,19 +433,16 @@ namespace
             }
         }
 
-        if (dynamic_pointer_cast<ClassDef>(p) || dynamic_pointer_cast<ClassDecl>(p) ||
-            dynamic_pointer_cast<Struct>(p) || dynamic_pointer_cast<Slice::Exception>(p))
-        {
-            string file = p->file();
-            assert(!file.empty());
-            DefinitionContextPtr dc = p->unit()->findDefinitionContext(file);
-            assert(dc);
+        string file = p->file();
+        assert(!file.empty());
+        DefinitionContextPtr dc = p->unit()->findDefinitionContext(file);
+        assert(dc);
 
-            // TODO: why do we ignore all instances of this metadata except the first?
-            if (auto headerFile = dc->getMetadataArgs("cpp:doxygen:include"))
-            {
-                out << nl << "/// \\headerfile " << *headerFile;
-            }
+        // The metadata directive is cpp:doxygen, not cpp:doxygen:include. The arg of cpp:doxygen is something like
```

Comment:
The previous code incorrectly referenced a metadata directive with an invalid syntax: `cpp:doxygen:include` is not and can't be a metadata directive.